opm/autodiff/RedistributeDataHandles.hpp

```diff
@@ -327,7 +325,8 @@ void distributeGridAndData( Dune::CpGrid& grid,
     grid.loadBalance(eclipseState);
     grid.switchToDistributedView();
 
-    distributed_props = std::make_shared<BlackoilPropsAdFromDeck>(properties, grid.numCells());
+    BlackoilPropsAdFromDeck distributed_props = BlackoilPropsAdFromDeck(properties,
+                                                                        grid.numCells());
     distributed_state.init(grid.numCells(), grid.numFaces(), state.numPhases());
```

Comment:
> handled with  4218a65 

Thanks.  Could you do the same transformation to `distributed_geology` on lines [341-343](https://github.com/blattms/opm-autodiff/blob/removed-superfluous-ptr-during-redistribution/opm/autodiff/RedistributeDataHandles.hpp#L341-L343)?  In other words, replace

```C++
DerivedGeology distributed_geology = DerivedGeology(/* args */);
```
by
```C++
DerivedGeology distributed_geology(/* args */);
```
?